src/components/schedule-page/corrupted-schedule.component.tsx

```diff
@@ -21,13 +22,31 @@ export function CorruptedScheduleComponent(): JSX.Element {
   const { past } = useSelector(
     (state: ApplicationStateModel) => state.actualState.persistentSchedule
   );
-  const isPreviousVersionAvailable = past.length > MINIMUM_UNDO_COUNT_TO_REVERT_NORMAL_SCHEDULE;
+  const { month_number: month, year } = useSelector(
+    (state: ApplicationStateModel) => state.actualState.persistentSchedule.present.schedule_info
+  );
+
+  const isCurrentNotCorruptedSchedule = useCallback(
+    (schedule: ScheduleDataModel): boolean => {
+      return (
+        !schedule.isCorrupted &&
+        schedule.schedule_info.month_number === month &&
+        schedule.schedule_info.year === year
+      );
+    },
+    [month, year]
+  );
+
+  const currentMonthPastRevisions = past.filter((schedule) =>
+    isCurrentNotCorruptedSchedule(schedule)
```

Comment:
Nie chcę być jak IDE, ale tutaj można uprościć XD
`past.filter(isCurrentNotCorruptedSchedule)`

Reply:
O faktycznie, zmienię to 😁